### ISSUE TITLE
Is it safe to delete files in LocalState\rootfs\mnt using Windows Explorer?

### ISSUE THREAD

**Issue by ZYinMD** [Microsoft/WSL#3737]
My SSD C: Drive is out of space, and I found some big temp files inside `C:\Users\MyUsername\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu18.04onWindows_79rhkp1fndgsc\LocalState\rootfs\mnt\D`. Is it safe to delete those temp files from Windows Explorer?

I'm asking because I was under the impression that I should never mess with WSL files from Windows.

Thanks!!

* Your Windows build number: 
 10.0.17763.195

**Comment by therealkenc** [comment]
If you reboot your machine and open the WSL shell again, do they disappear? Asking because it self-sorts the answer. (a) No, you shouldn't mess around in there. (b) You kind of have to mess around in there absent any other options. And rebooting is about your only other option; ensuring there no "state" you could possibly care about at that point.

That path is weird though. The path `...\rootfs\mnt\D` should just be a mountpoint, and should be empty fullstop, AFAIK. It's also odd the capital 'D' since that isn't the usual default (it would be '`d`'). What's lacking here is how those files got there in the first place.

If you can grab a screencap to see what these "temp" files look like that might give some context.

**Comment by ZYinMD** [comment]
Thanks @therealkenc !!

Sorry the driver letter was indeed `d`, not `D`, my typo.

Those temp files were generated by [youtube-dl](https://github.com/rg3/youtube-dl), which was a CLI tool for downloading videos, and the temp files are actually unfinished downloads like `...\LocalState\rootfs\mnt\d\videos\foo.mp4`. I guess it writes to d: when the download finishes, but when some force-close happens, it's unable to do that, thus leaving the partial downloads behind. Over time I've accumulated about 10G of them.

I rebooted as you said, and they're still there, so I guess my next option is to just delete them by hand. I'm 100% sure I don't need them.

**Comment by therealkenc** [comment]
Thanks for the follow up. I'll let this one float a bit, because I'm still baffled by the location of those files (aborted youtube scrapes or otherwise). That's not on you. I'm just confused a bit is all. I suspect you've found a real issue, but as it stands neither of us know how they got there. My `.....\mnt\c` is clean, but that's not newsworthy. Delete away.

**Comment by ZYinMD** [comment]
Well, since you think it's a real bug, I don't want to give wrong information, so I just took a closer look.

I found that I didn't actually "accumulate the files over time". Those files were actually created on the same day (2 months ago), within a period of several hours. Since it can only download one video at a time, the scraping must have been continuously running on that day, so WSL didn't force close for sure, although the scraping itself may have aborted some files.

I run my scraping like this:
`youtube-dl --output '/mnt/d/Video Archive/%(uploader)s/%(title)s.%(ext)s' --batch-file MyURLs.txt --ignore-errors`
explanation:
`--output`: the folder structure to store the videos
`--batch-file`:  the file that contains video urls to be downloaded. It supports playlists and channels, so I have lots of those.
`--ignore-errors`: if one video throws an error, skip to the next.

That day was not the only time I run it. I do this quite often, with the same settings.

All the files left in `rootfs\mnt` maintained their folder structure, for instance `...\LocalState\rootfs\mnt\d\Video Archive\Uploader37\VideoTitle258.mp4`

Some other info that is probably irrelevant:
1. My default user in wsl is `root`.
2. My d: is a usb drive. Which may had a bad connection on that day? But I'm sure Windows didn't alert me "improperly unplugged", unless it can plug itself back in.
3. The drive letter is actually not d:, it's actually t:, I re-assigned this letter using Disk Management. I said d: only to simplify my question.
